modules/displayioBidAdapter.js

```diff
@@ -42,117 +45,112 @@ export const spec = {
-        mediation.consent = gdprConsent.consentString;
-      }
-      if (gdprConsent.gdprApplies !== undefined) {
-        mediation.gdpr = gdprConsent.gdprApplies ? '1' : '0';
-      }
+  }
+};
+
+function getPayload (bid, bidderRequest) {
+  const connection = navigator.connection || navigator.mozConnection || navigator.webkitConnection;
+  const userSession = 'us_web_xxxxxxxxxxxx'.replace(/[x]/g, c => {
+    let r = Math.random() * 16 | 0;
+    let v = c === 'x' ? r : (r & 0x3 | 0x8);
+    return v.toString(16);
+  });
+  const { params, adUnitCode, bidId } = bid;
+  const { siteId, placementId, renderURL, pageCategory, keywords } = params;
+  const { refererInfo, uspConsent, gdprConsent } = bidderRequest;
+  const mediation = {consent: '-1', gdpr: '-1'};
+  if (gdprConsent && 'gdprApplies' in gdprConsent) {
+    if (gdprConsent.consentString !== undefined) {
+      mediation.consent = gdprConsent.consentString;
     }
-    const payload = {
-      userSession,
+    if (gdprConsent.gdprApplies !== undefined) {
+      mediation.gdpr = gdprConsent.gdprApplies ? '1' : '0';
+    }
+  }
+  return {
+    userSession,
+    data: {
+      id: bidId,
+      action: 'getPlacement',
+      app: siteId,
+      placement: placementId,
+      adUnitCode,
+      renderURL,
       data: {
-        id: bid.bidId,
-        action: 'getPlacement',
-        app: siteId,
-        placement: placementId,
-        data: {
-          pagecat: params.pageCategory ? params.pageCategory.split(',').map(k => k.trim()) : [],
-          keywords: params.keywords ? params.keywords.split(',').map(k => k.trim()) : [],
-          lang_content: document.documentElement.lang,
-          lang: window.navigator.language,
-          // TODO: are these the correct refererInfo values?
-          domain: refererInfo.domain,
-          page: refererInfo.page,
-          ref: refererInfo.ref,
-          userids: _getUserIDs(),
-          geo: '',
-        },
-        complianceData: {
-          child: '-1',
-          us_privacy: uspConsent,
-          dnt: window.navigator.doNotTrack,
-          iabConsent: {},
-          mediation: {
-            consent: mediation.consent,
-            gdpr: mediation.gdpr,
-          }
-        },
-        integration: 'JS',
-        omidpn: 'Displayio',
-        mediationPlatform: 0,
-        prebidVersion: BIDDER_VERSION,
-        device: {
-          w: window.screen.width,
-          h: window.screen.height,
-          connection_type: connection ? connection.effectiveType : '',
+        pagecat: pageCategory ? pageCategory.split(',').map(k => k.trim()) : [],
+        keywords: keywords ? keywords.split(',').map(k => k.trim()) : [],
+        lang_content: document.documentElement.lang,
+        lang: window.navigator.language,
+        domain: window.location.hostname,
+        page: window.location.href,
+        ref: refererInfo.referer,
+        userids: deepAccess(bid, 'userId') ? createEidsArray(bid.userId) : {},
```

Comment:
you shouldn't need to import createEidsArray. The output of this function is already on the bid request

Reply:
@patmmccann Thank you for help! fixed